vector/src/main/java/im/vector/app/core/preference/KeywordPreference.kt

```diff
@@ -0,0 +1,125 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.preference
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.inputmethod.EditorInfo
+import android.widget.EditText
+import androidx.core.view.children
+import androidx.preference.PreferenceViewHolder
+import com.google.android.material.chip.Chip
+import com.google.android.material.chip.ChipGroup
+import im.vector.app.R
+
+class KeywordPreference : VectorPreference {
+
+    interface Listener {
+        fun didAddKeyword(keyword: String)
+        fun didRemoveKeyword(keyword: String)
+    }
+
+    private var keywordsEnabled = true
+
+    private var _keywords: LinkedHashSet<String> = linkedSetOf()
+
+    var keywords: Set<String>
+        get() {
+            return _keywords
+        }
+        set(value) {
+            // Updates existing `LinkedHashSet` vs assign a new set.
+            // This preserves the order added while on the screen (avoids keywords jumping around).
+            _keywords.removeAll(_keywords.filter { !value.contains(it) })
+            _keywords.addAll(value.sorted())
+            notifyChanged()
+        }
+
+    var listener: Listener? = null
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
+
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)
+
+    init {
+        layoutResource = R.layout.vector_preference_chip_group
+    }
+
+    override fun setEnabled(enabled: Boolean) {
+        super.setEnabled(enabled)
+        keywordsEnabled = enabled
+        notifyChanged()
+    }
+
+    override fun onBindViewHolder(holder: PreferenceViewHolder) {
+        super.onBindViewHolder(holder)
+
+        holder.itemView.setOnClickListener(null)
+        holder.itemView.setOnLongClickListener(null)
+
+        val chipEditText = holder.findViewById(R.id.chipEditText) as? EditText ?: return
+        val chipGroup = holder.findViewById(R.id.chipGroup) as? ChipGroup ?: return
+
+        chipEditText.text = null
+        chipGroup.removeAllViews()
+
+        keywords.forEach {
+            addChipToGroup(it, chipGroup)
+        }
+
+        chipEditText.isEnabled = keywordsEnabled
+        chipGroup.isEnabled = keywordsEnabled
+        chipGroup.children.forEach { it.isEnabled = keywordsEnabled }
+
+        chipEditText.setOnEditorActionListener { _, actionId, _ ->
```

Comment:
Would be nice to also have a button "Add" below the edit text. I'm afraid the IME_ACTION can be missed be some users.

Reply:
Wasn't included in the designs/don't think they felt it was necessary, but I see where you are coming from.
Maybe we can get feedback and see if it needs to be added. But will run it by design when they are available either way.